### ISSUE TITLE
RawSourceWaveStream.CurrentTime does not set the value properly.

### ISSUE THREAD

**Issue by siaf** [naudio/NAudio#106]
It appears NAudio changes the position incorrectly to 1 index before where it should be

In this example I have a RawStream of PCM 16Bit8Khz audio.
Since this is 16bit audio each sample is 2 bytes.If this is not on the write place

```
rawStream.CurrentTime = newTimeSpan;
if (rawStream.Position % 2 != 0)
{
    rawStream.Position = rawStream.Position + 1;
}
```

So for example if I set it to 10 Milliseconds (assuming this is one of those cases) 
Instead of position being 160 it might become 159. 

I've notices in the CurrentTime property for WaveStream there's this line:

`
   Position = (long) (value.TotalSeconds * WaveFormat.AverageBytesPerSecond);`

As you can see currently I'm using a hack to check for the position if it's proper and if not I'll fix it.
Is this by design? is there any way to do this better?

Thanks

**Comment by markheath** [comment]
Yes, when you reposition you should take account of the blockalign. Many of the classes in NAudio do this for you in their position setter. I guess it would be good if the RawSourceStream did this too

**Comment by hozuki** [comment]
I noticed this problem too. I wrote a custom wave stream provider to dynamically decode audio in another format to raw wave format. I'm sure the seeking function in the custom provider is correct, but when I try to seek through a track in a mini player, there is a good chance to output noise. Then I found the `BlockAlign` property, and traced back to [the same line](https://github.com/naudio/NAudio/blob/f98664ba3d5966a736dd5cf45329e542be5fece8/NAudio/Wave/WaveStreams/WaveStream.cs#L113). In my case the alignment is 4 (stereo, 16-bit), and about a half of seeking trials end in noise.

I think `WaveStream` class should perform the checking, rather than letting subclasses to do so.

And thanks for the `Position` hack!

**Comment by tweigt** [comment]
Should this not be fixed in `WaveStream.CurrentTime` setter?
I have changed it like this:

    Position = (long)(value.TotalSeconds * WaveFormat.SampleRate) * WaveFormat.BlockAlign;

You may optionally add a call to `Math.Round` before the cast to `long`.
By separating the multiplication like this, you are guaranteed to be on the boundary of a sample, regardless of the value of `TotalSeconds` and no modulo needed.
